[WIP] Website refactoring follow-up

mxsasha opened the pull request:
Carried over from #1010:
- [x] Have validation/schema for all yaml files (sessions and schedule are not checked <2020 as we're unlikely to still touch them often)
- [x] Review referenced tickets from #1010 (#872, #658)
- [x] Make sure #901 still works
- [x] Check #1065 against new and old conference schedule setup
- [x] See whether we can remove duplication of menu templates (related #492)
- [x] See if docs/include and docs/_templates/include should be merged/reorganised
- [x] Documentation for conf websites (including #681)
- [x] Look at splitting off some of the sponsorship data into includes (packages are per conference) -- already done for 2020 portland
- [x] Consider including 2013 and 2014 websites, currently on github pages
- [x] Add a workflow for "tickets sold out"

**Do not use the github button to fix this branch being outdated compared to master, because it makes rebasing and cleanup very hard.**

mxsasha committed:
Update config YAML validation and enable strict mode, fix #827

YAML validation will now fail when encountering unknown keys.
The `about.sat` setting was removed from Australia 2018/2019 (where it
was empty) and Portland 2018 (where it was `5`). It seems unused.
The `about.videos` setting was removed from Prague 2018, because it
is never actually used.

mxsasha committed:
Add validation for >=2020 sessions and schedule

mxsasha committed:
Fix #658 - Add projector ratio to conference config.

mxsasha committed:
Ref #492 - Make menu templates conference-agnostic

mxsasha commented:
I have attempted to move menu's to the YAML, as suggested in #1010, to fix #492. However, it's more complex than it seems, and helps poorly in making it easier to edit. Instead, I improved upon our menu situation by making the 2020 menu pages conference-agnostic (added `flaghasbadgeflair` for that). They should be year-agnostic too, but let's review that for 2021.

mxsasha committed:
Ref #901 - Add run of show output/template for 2020

mxsasha committed:
Fix confusing naming in 2020 schedule template

mxsasha commented:
Regarding [include directories](https://github.com/writethedocs/www/pull/1010#issuecomment-520166152), I reviewed them and the current layout makes sense to me:
* templates/_include/ contains template includes, used to render content, like the snippet used to render a schedule, or the run of show, but no actual content
* include/ contains actual content that we reuse in different conference website

I did not find any other include directories.

mxsasha commented:
I looked into including the 2013/2014 conference sites, and I don't think it's worth it. It's seems pretty messy to add all the legacy css/js/img, and also all of the resource links would have to be fixed. As it takes no effort to keep the old github pages running, I'm inclined to keep things as they are.

mxsasha committed:
Add documentation for the conference website, fix #681.

plaindocs commented on docs/_data/config-portland-2018.yaml:
```diff
@@ -28,7 +28,6 @@ date: # how do we handle these? human readable would be nice.
   short: May 6-8, 2018
   tickets_live: "January 2018"
   month: May
```
Do we want to try to add some metadata that would let us fix this? https://github.com/writethedocs/www/pull/1125/files#diff-66dfabd3b4a41d17c53d19ac40e964c6R84

plaindocs commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,252 @@
+
+.. highlight:: yaml
+
+Conference websites
```
:heart: :100:

mxsasha committed:
Ref #1065 - Remove schedule-csv2yaml script.

With the simplification of conference schedules, the
plan is to simply write the YAML directly, or rather
copy from last year and adjust.

mxsasha committed:
Add and document a "sold out" state for confs

mxsasha commented:
For review: this isn’t as massive as #1010, but it might still be easiest to review individual commits rather than the whole thing, as they’re not all very related. Also note that the YAML validation commit includes cleaning up obsolete or never used settings, as we now use strict mode meaning you can’t add random unknown attributes in the yaml anymore.

plaindocs reviewed (approved):
I'm :+1: on all of this. 

We *should* fix the date thing once and for all, but happy to do that after this so we don't block.

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+===================
+
+Each conference has a section of the website, which is part of the same
+repository as the main conference website. Many parts of the conference
```
```suggestion
repository as the main community website. Many parts of the conference
```

thatdocslady replied:
The sentence currently reads: "Each conference has a section of the website, which is part of the same repository as the main conference website."

Usage of the word "conference" confuses me, so I went with a suggestion based on what seemed logical in my head. Our www repository has the main community website, and a directory for each conference, so that's what seemed to make sense. I fixed the typo ;)

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+
+* ``_data/<shortcode>-<year>-config.yaml`` contains the general config file
+  for rendering a specific conference. This includes settings like the
+  conference dates, sponsors and ticket prices.
```
```suggestion
  conference dates, sponsors, and ticket prices.
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  the main talks on the agenda. For each session, it includes things like the
+  title and abstract, and name and other info for the speaker(s).
+* ``_data/<shortcode>-<year>-schedule.yaml`` contains the conference schedule.
+  This includes the talks, but also items like "Lunch".
```
```suggestion
  This includes the talks, but also peripheral activities like lightning talks, meals, the outing, and social event.
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+* ``_data/<shortcode>-<year>-schedule.yaml`` contains the conference schedule.
+  This includes the talks, but also items like "Lunch".
+* ``conf/<shortcode>/<year>/`` contains the RST files used for rendering the
+  conference website. Copy these over from the previous conference
```
This is the only list item where we ask the readers to do something. Since we are documenting the website workflow later, it's probably better to keep this list as a reference and clean out task-style content. It confused me and will probably confuse others...

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  conference website. Copy these over from the previous conference
+  (chronologically, not the previous conference in the same location).
+  You may need to update some reference to specific locations/years,
+  and keep an eye on any FIXME's or TODO's.
```
```suggestion
  and keep an eye on any FIXMEs or TODOs.
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  (chronologically, not the previous conference in the same location).
+  You may need to update some reference to specific locations/years,
+  and keep an eye on any FIXME's or TODO's.
+* ``include/conf/`` contains text snippets which are generally the same
```
```suggestion
* ``include/conf/`` contains text snippets that are generally the same
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  between conferences, or at least between conferences in the same location.
+  Examples include the description of lightning talks, or the Portland hike.
+* ``_templates/<year>/`` and ``_templates/include/`` contain HTML or RST
+  templates used in rendering the conference pages. This includes the
```
```suggestion
  templates used to render the conference pages. This includes the
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  templates used in rendering the conference pages. This includes the
+  website design, navigation and snippets used in schedule rendering.
+
+The YAML files are validated using the schemas in ``_data/schema-*``.
```
Passive voice alert! Who/what is doing the validation? Is it something a human needs to do or is it automated?

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+The YAML files are validated using the schemas in ``_data/schema-*``.
+
+
+Typical conference workflow
```
Mega-passive-voice alert! We should rewrite this section to make it clear which tasks are done by humans and which ones are automated. We should use second-person voice for tasks that human do, and third-person for automated things, like "The RTD CI builds the website after you merge a PR"...

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+
+* The initial conference site is created, with ``flaglanding`` typically
+  set to publish a simple landing page with the conference dates.
+* At some point, the content for the conference is filled in and verified,
```
```suggestion
* At some point before the full website launch, the content for the conference is filled in and verified,
```

thatdocslady replied:
We should probably have people review the activities in the conference to see if anything needs to be updated (mostly times/locations that aren't in variables and a quick skim of the team page and reusable snippets)

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  like sponsorship prices. ``flaglanding`` is now switched off and remains off.
+* As soon as tickets go on sale, ``flagticketsonsale`` is switched on.
+* When the CFP opens, ``flagcfp`` is switched on. Once the CFP has closed,
+  it is switched off and remains off. The CFP details should be in the
```
"should" is a name of a fish ;) does this mean that the CFP details can be found in this file, or that people must add them to the file?

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  it is switched off and remains off. The CFP details should be in the
+  general conference config file, under ``cfp``.
+* When speakers are announced, ``flagspeakersannounced`` is switched on
+  and remains on. The sessions should be in the
```
"should" alert 2 (fix everywhere)

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  The schedule should be in the ``_data/<shortcode>-<year>-schedule.yaml`` file.
+* At some point the conference sells out. ``flagticketsonsale`` is switched off,
+  ``flagsoldout`` is switched on.
+* During the conference, if live streaming is available, ``flaglivestreaming``
```
This needs to happen before the conference, not during, so that we can promote the live stream (1-2 weeks?)

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  is switched on and remains on.
+
+Each time the state is updated, check the top/bottom buttons to ensure they
+still fit. For example, once tickets have sold out, the button "buy a ticket"
```
Fit = text doesn't overrun? Can we state a character limit?

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+
+The items in the general conference config file are:
+
+* ``name``: the dynamic part of the name of the conference, e.g. ``Portland``
```
City name?

mxsasha replied:
Not always, for Australia this would be "Australia".

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  ``portland``, ``prague``, and ``australia``.
+* ``year``: the conference year, e.g. ``2020``.
+* ``city``: the city for the conference.
+* ``local_area``: the state or country for the conference.
```
For example, "Oregon" or "Czech Repubilc"? Does this need to be properly capitalized?

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+  references to the conference, together with the year. Currently used are
+  ``portland``, ``prague``, and ``australia``.
+* ``year``: the conference year, e.g. ``2020``.
+* ``city``: the city for the conference.
```
```suggestion
* ``city``: the city for the conference, such as "Portland" or "Prague". For Australia, use "Australia".
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+* ``area``: the wider area for the conference. Currently used are
+  ``North America``, ``Europe``, and ``Australia``.
+* ``area_adj``: the adjective form of the area, e.g. ``Australian``.
+* ``tz``: the timezone the conference is located in.
```
Format?

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+* ``tz``: the timezone the conference is located in.
+* ``email``: general contact email for the conference.
+* ``color``: the accent color for conference branding, e.g. ``blue`` for 2020.
+* ``photos.default``: the default header image.
```
How do we choose the picture? Is that OOS for this role?

mxsasha replied:
I'd say it's out of scope. We can add that later if we have a process for it.

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+      top:
+         - text: Sponsor the conference
+           link: /sponsors/prospectus
+         - text: Submit a Talk
```
Is capitalization important?

mxsasha replied:
Maybe. This is the historical pattern. In the interface they are shown all-caps, but I'm not sure whether capitalisation matters for screen readers.

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+      student:
+        price: $100
+
+* ``sponsorship``: the conference sponsorship prices. The keys are fixed and
```
what does "fixed" mean? the amounts for tix/sponsorships change between conferences and years sometimes, but it's something that needs to be decided by consensus.

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+    * ``main``: the human readable conference dates and location,
+      e.g. ``"**May 3-5, 2020, in Portland, Oregon**"``.
+    * ``short``: the short human readable dates, e.g. ``May 3-5, 2020``.
+    * ``tickets_live``: a rough human readable date when tickets go on sale,
```
```suggestion
    * ``tickets_live``: a human readable date to indicate the month when tickets go on sale,
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+    * ``photos``: a link to the conference photos (typically on Flickr).
+    * ``mainroom``: the name of the main room.
+    * ``unconfroom``: the name of the unconference room.
+    * ``projector_ratio``: the ratio for the projector, e.g. ``16:9``.
```
this seems like it should belong in another section, do we have/can we make a speaker section, or add this to the CFP?

mxsasha replied:
I'm reluctant to add a section for 1 config item, CFP also doesn't feel like it covers it? I'm inclined to keep it where it is.

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+      e.g. ``31 January, 2020``.
+    * ``notification``: a human readable date of when accepted speakers
+      will be notified.
+* ``job_fair.location``: a description of the location of the job fair.
```
Example? Is this just a room name? Maybe move this to the section that shows the other rooms....

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+      will be notified.
+* ``job_fair.location``: a description of the location of the job fair.
+
+Then, there are a number of true/false flags. Some of these don't change
```
```suggestion
The file also includes a few true/false flags. Some of these don't change
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+* ``flaghasschedule``: is the schedule ready?
+  This flag requires the``_data/<shortcode>-<year>-schedule.yaml``
+  file to exist, which contains the schedule.
+* ``flaglivestreaming``: is a live stream currently running?
```
probably should change this to "is a live stream available", since the page needs to go live before we start the conf..

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+Sessions file
+-------------
+
+The sessions file contains the conference sessions, i.e. the talks.
```
Maybe use path/filename instead of just "sessions" file

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+
+* ``title``: title of the talk
+* ``abstract``: talk abstract
+* ``slug``: talk slug - referenced in the schedule.Typically, the slug is
```
```suggestion
* ``slug``: talk slug - referenced in the schedule. Typically, the slug is
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+* ``title``: title of the talk
+* ``abstract``: talk abstract
+* ``slug``: talk slug - referenced in the schedule.Typically, the slug is
+  a slugified version of the title, followed by the slugified speaker name.
```
slugified? what is that?

mxsasha replied:
It's essentially translating the title into lowercase, removing any non-letters, replacing spaces by dashes. It's a fairly widely used term, but there is no universally accepted meaning. I can link to something like https://blog.tersmitten.nl/slugify/ ?

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+* ``abstract``: talk abstract
+* ``slug``: talk slug - referenced in the schedule.Typically, the slug is
+  a slugified version of the title, followed by the slugified speaker name.
+* ``series``: the conference series, used for videos, e.g. ``Write the Docs PORTLAND``
```
1 - why is PORTLAND all-caps?
2 - what does series mean? is it referring to which conference it belongs to?

mxsasha replied:
1 - historical reasons, something we should pick up when improving the video system.
2 - I think so? But all the video stuff is a bit misty.

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+* ``year``: the year the talk was given, used for videos
+* ``youtubeId``: the Youtube ID of the talk video, if published already
+* ``speakers``: the speaker(s) for the talk, in a list of speakers with keys:
+    * ``name``: name of the speaker
```
```suggestion
    * ``name``: full name of the speaker
```

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+the schedule file of the same conference as last year, as they tend to be
+quite similar.
+
+For conferences with a writing day, there must be a ``writing_day`` key. Then,
```
```suggestion
For conferences with a writing day, you must add a ``writing_day`` key. Then,
```

thatdocslady replied:
where do you add this key to?

thatdocslady replied:
Omg I see what "key" means. Maybe we can define "key" somewhere, I don't see the term used so much so other mortals might need this clarified.

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+quite similar.
+
+For conferences with a writing day, there must be a ``writing_day`` key. Then,
+a schedule for each main conference day, in the form of ``day1``, ``day2``,
```
in one of the previous files, "day1" and so on is used to indicate whatever days, so for example day 1 would be the hike/boat, but in the schedule file day1 means day 1 of talks and things. This might cause some confusion, not sure how we can fix this... maybe standardize on the day numbering so that we all know what day 1 means everywhere?

mxsasha replied:
This is difficult, in part due to legacy, but I will change the schedule, which is still very new, to say "talks-day1" etc. That'll reduce some of the confusion.

thatdocslady commented on docs/organizer-guide/confs/website.rst:
```diff
@@ -0,0 +1,293 @@
+      - time: '9:40'
+        title: "<b>Unconference Starts (Lola's Room)</b>"
+      - time: '9:40'
+        slug: any-friend-of-the-docs-is-a-friend-of-mine-cultivating-a-community-of-documentation-advocates-heather-stenson
```
is this a copy-paste operation?

mxsasha replied:
The slugs are, yes. The file as a whole too, as a start, because we have almost identical schedules each year per conference. Clarified that elsewhere.

thatdocslady reviewed (commented):
Docs look great, bunch of nit picks ;)

mxsasha committed:
Various documentation improvements.

Co-Authored-By: Mikey Ariel <thatdocslady@users.noreply.github.com>

mxsasha committed:
Change keys in schedule from day1 to talks-day1 to reduce confusion.

mxsasha committed:
Move job fair room to the same place as other room names.

mxsasha commented:
@thatdocslady almost all your comments are addressed, if you can have a look at some of my responses on the non-resolved ones when you have time, we can wrap this up :)

mxsasha committed:
Fix generic mobile menu link
